### ISSUE TITLE
In social feed, attachment titles & previews don’t link to attachment itself

### ISSUE THREAD

**Issue by bfackland** [ploneintranet/ploneintranet#848]
![screen shot 2015-10-22 at 11 09 20](https://cloud.githubusercontent.com/assets/2538526/10662558/83a7732c-78ad-11e5-9f73-c5cfcfaa5ebd.png)

**Comment by mattss** [comment]
Appears to just be missing some <a> tags. The prototype clearly links both image and title to the same place as the 'download' arrow.

**Comment by wengole** [comment]
Can't make the image clickable due to the unique way in which Plone Intranet uses patterns (i.e. it uses patterns)

**Comment by wengole** [comment]
Ok that's not quite true, but the "preview unavailable" CSS can't be made clickable.

**Comment by bfackland** [comment]
I guess we can live with that for now - no preview = no link from the placeholder image.

The majority of the time we _will_ have a preview, so it's important that image links to the full document (whatever that is).

**Comment by bfackland** [comment]
In another task, @gyst wrote:

> The title and preview are not linked, and there is no 'view' icon: this is because the attachment does not have a content life of it's own, it exists only as an appendix to this statusupdate

In this task have explicitly asked for the title and image to link to the same place as the download icon, which @wengole has implemented. @gyst do you have concerns with this?

**Comment by gyst** [comment]
@cornae has the vision to explicitly distinguish between actual content updates (i.e. persistent content hosted elsewhere in the system showing up in the stream) and social postings with a content object attached in the stream only (i.e. transient sharing). The key here is the difference between 'view' and 'download' where the view action shows you a HTML page and the download action gives you some binary. The title, preview and view icon all link to the view action, only the download button links to the download action. Since stream attachments do not have a location other than the stream update itself, they do not have a view action, only a download action. Linking the title and preview there to download breaks UX consistency with the content updates, which do support viewing.

**Comment by gyst** [comment]
I.e. IMO #860 should be reverted. I did a lot of work earlier to clean this up and make it conform to prototype...

**Comment by cornae** [comment]
That's correct. Clicking on the preview should only link to the object in a workspace if such an object exists. What is still missing in the implementation is the eye icon. The eye-icon triggers a full screen preview that you can swipe, click or key through.  CSS and .js and the markup examples for all different uses cases are available in the prototype on the dashboard timeline. Bear in mind that the images in proto are random lorem pixel images, not proper demo data of actual office documents, by lack of budget to create the demo date. Could be confusing if you don't know.

![image](https://cloud.githubusercontent.com/assets/738601/10788146/c211aea8-7d75-11e5-8f27-0ed7f0aa429b.png)

**Comment by cornae** [comment]
I see actually also the page count icon is still missing. It's a useful feature that triggers to preview if applicable.

**Comment by gyst** [comment]
I don't think the multi-preview feature has been fully implemented yet, @pilz knows the score.
